src/main/java/duke/component/DukeDatabase.java

```diff
@@ -0,0 +1,144 @@
+package duke.component;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Encapsulates the database of Duke bot.
+ */
+public class DukeDatabase {
+    private String databaseDirectory; // Path of the the database file.
+    private File tasksFile; // File object which represents the database file of Duke bot
+
+    // Cannot delete this constructor to ensure that it overwrites the default public constructor
+    private DukeDatabase() {
+    }
+
+    /**
+     * Returns a DukeDatabase object.
+     *
+     * @return a DukeDatabase object.
+     */
+    public static DukeDatabase getDukeDatabaseInstance() {
+        return new DukeDatabase().initialise();
+    }
+
+    /**
+     * Initializes the database by loading the database file and attach it to a File object.
+     *
+     * @return the database object itself.
+     */
+    private DukeDatabase initialise() {
+        // Generates the directory for the database
+        String projectRoot = new File(System.getProperty("user.dir"))
+                .getParentFile()
+                .getPath();
+        databaseDirectory = String.format("%s/data/duke.txt", projectRoot);
+
```

Comment:
I would add the following here:

```
if (tasksFile.exists()) {
   return;
}
```

It appears that if the file exists, IOException will always occur on line 51. After checking Java Documentation, it actually does not throw an exception. The exist check happens atomically with all other filesystem operations. From a readability standpoint, it would be clearer to read with the above snippet.